symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/user/UserService.java

```diff
@@ -0,0 +1,520 @@
+   */
+  public List<V2UserDetail> listUsersDetail() {
+    try {
+      return userApi.v2AdminUserListGet(authSession.getSessionToken(), null, null);
+
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Retrieve a list of users in the company (pod) by a filter.
+   *
+   * @param filter  using to filter users by
+   * @return        List of retrieved users
+   * @see           <a href="https://developers.symphony.com/restapi/reference#find-users">Find Users V1</a>
+   */
+  public List<V2UserDetail> listUsersDetail(@NonNull UserFilter filter) {
+    try {
+      List<UserDetail> userDetailList = userApi.v1AdminUserFindPost(authSession.getSessionToken(), filter, null, null);
+      return userDetailList.stream().map(UserDetailMapper.INSTANCE::userDetailToV2UserDetail).collect(Collectors.toList());
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Add a role to an user.
+   *
+   * @param uid     User Id
+   * @param roleId  Role Id
+   * @see           <a href="https://developers.symphony.com/restapi/reference#add-role">Add Role</a>
+   */
+  public void addRoleToUser(@NonNull Long uid, @NonNull RoleId roleId) {
+    try {
+      StringId stringId = new StringId().id(roleId.name());
+      userApi.v1AdminUserUidRolesAddPost(authSession.getSessionToken(), uid, stringId);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Remove a role from an user.
+   *
+   * @param uid     User Id
+   * @param roleId  Role Id
+   * @see           <a href="https://developers.symphony.com/restapi/reference#remove-role">Remove Role</a>
+   */
+  public void removeRoleFromUser(@NonNull Long uid, @NonNull RoleId roleId) {
+    try {
+      StringId stringId = new StringId().id(roleId.name());
+      userApi.v1AdminUserUidRolesRemovePost(authSession.getSessionToken(), uid, stringId);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Get the url of avatar of an user
+   *
+   * @param uid User Id
+   * @return    List of avatar urls of the user
+   * @see       <a href="https://developers.symphony.com/restapi/reference#user-avatar">User Avatar</a>
+   */
+  public List<Avatar> getAvatarFromUser(@NonNull Long uid) {
+    try {
+      return userApi.v1AdminUserUidAvatarGet(authSession.getSessionToken(), uid);
+    } catch (ApiException apiException) {
+      throw new ApiRuntimeException(apiException);
+    }
+  }
+
+  /**
+   * Update avatar of an user
+   *
+   * @param uid   User Id
+   * @param image The avatar image for the user profile picture.The image must be a base64-encoded.
+   * @see         <a href="https://developers.symphony.com/restapi/reference#update-user-avatar">Update User Avatar</a>
+   */
+  public void updateAvatarOfUser(@NonNull Long uid, @NonNull String image) {
```

Comment:
As a API user in Java I would perhaps prefer to send an inputstream here (and then have the SDK do the encoding for me)